File: web/src/types/debates.ts
export interface CreateDebateRequest {
  name?: string | null
  symbol?: string | null
  max_rounds?: number | null
  prompt_variant?: string | null
  participants?: string[] | null
}

export interface StartDebateRequest {
  model_ids?: string[] | null
}

export interface DebateSummaryPayload {
  id: string
  name: string
  symbol: string
  status: string
  max_rounds: number
  current_round: number
  final_decision: string
  created_at: number
}

export interface DebatePayload extends DebateSummaryPayload {
  prompt_variant: string
  participants: string[]
  final_reasoning: string
  error_message: string
  updated_at: number
}

export interface DebateRoundVotePayload {
  personality: string
  vote: string
}

export interface DebateVoteRoundPayload {
  round: number
  votes: DebateRoundVotePayload[]
  tally: Record<string, number>
}

export interface DebateListPayload {
  debates: DebateSummaryPayload[]
  count: number
}

export interface DebateDetailPayload {
  debate: DebatePayload
}

export interface DebateActionPayload {
  id: string
  message: string
}

export interface DebateMessagePayload {
  message: string
}

export interface DebatePersonalityPayload {
  id: string
  name: string
  description: string
}

export interface DebatePersonalitiesPayload {
  personalities: DebatePersonalityPayload[]
}

export interface DebateExecutionPayload {
  id: string
  final_decision: string
  final_reasoning: string
  message: string
}

export interface DebateMessagesPayload {
  messages: DebateMessageItemPayload[]
  count: number
}

export interface DebateMessageItemPayload {
  id: string
  round: number
  personality: string
  role: string
  content: string
  vote: string
  created_at: number
}

export interface DebateVotesPayload {
  votes: DebateVoteRoundPayload[]
}

File: core/src/contracts/debates.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Deserialize)]
pub struct CreateDebateRequest {
    pub name: Option<String>,
    pub symbol: Option<String>,
    pub max_rounds: Option<i64>,
    pub prompt_variant: Option<String>,
    pub participants: Option<Vec<String>>,
}

#[derive(Debug, Deserialize)]
pub struct StartDebateRequest {
    #[allow(dead_code)]
    pub model_ids: Option<Vec<String>>,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateListPayload {
    pub debates: Vec<Value>,
    pub count: usize,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateDetailPayload {
    pub debate: Value,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateActionPayload {
    pub id: String,
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateMessagePayload {
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebatePersonalityPayload {
    pub id: &'static str,
    pub name: &'static str,
    pub description: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebatePersonalitiesPayload {
    pub personalities: Vec<DebatePersonalityPayload>,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateExecutionPayload {
    pub id: String,
    pub final_decision: String,
    pub final_reasoning: String,
    pub message: &'static str,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateMessagesPayload {
    pub messages: Vec<Value>,
    pub count: usize,
}

#[derive(Debug, Clone, Serialize)]
pub struct DebateVotesPayload {
    pub votes: Value,
}
